src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -53,12 +53,11 @@
 
 _AUTH_CTX_FACTORY = lambda authority, cache: adal.AuthenticationContext(authority, cache=cache)
 
-def _read_file_content(file_path):
-    file_text = None
+def _load_tokens_from_file(file_path):
+    all_entries = []
     if os.path.isfile(file_path):
-        with codecs_open(file_path, 'r', encoding='ascii') as file_to_read:
-            file_text = file_to_read.read()
-    return file_text
+        all_entries = get_file_json(file_path, no_throw_on_empty=True) or []
```

Comment:
I reverted this chunk of change which has handled the reading pretty well, be utf+bom, utf, or ansi.